### ISSUE TITLE
Allow evaluation of Chebyshev polynomials outside of their domain

### ISSUE THREAD

**Issue by dancsi** [JuliaMath/Polynomials.jl#326]
Consider the following example:
```julia
using Polynomials
p = ChebyshevT([1])
p(2)
```
Currently it throws the following error:
```julia
ERROR: 2 outside of domain
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] (::ChebyshevT{Int64})(x::Int64)
   @ Polynomials ~/.julia/packages/Polynomials/1aa8e/src/polynomials/ChebyshevT.jl:96
 [3] top-level scope
   @ REPL[1]:3
```
I understand that most of the time evaluating T_n(x) outside [-1, 1] is indeed an error, but there are some cases where it is not (e.g. when implementing the Chebyshev iteration inverse polynomial, one has to evaluate T_n(1 + delta) for delta > 0). It would be nice to have an option to ignore the domain check. 

As an alternative to this, I tried converting the polynomial to the monomial basis and evaluating it there, but this gets ill-conditioned quite quickly. Finally, I sidestepped the issue by implementing the Clenshaw algorithm myself, since it's already implemented in Polynomials.jl.

**Comment by jverzani** [comment]
Hmm, no reason not. In `SpecialPolynomial`, this isn't the restriction. I'll make this available through a third argument, checked, to `evalpoly`, which for version v1.4 and above is an export so should be as direct as call `evalpoly(x, p, false)`.

**Comment by dancsi** [comment]
That was very fast, thanks!

**Comment by jverzani** [comment]
No problem. One thing is that I can't really backport this fix to the 1.2.X versions, as exposing `evalpoly` was done starting with v2.0.0.
